common/FileSystem.cpp

```diff
@@ -284,6 +284,107 @@ bool Path::IsAbsolute(const std::string_view path)
 #endif
 }
 
+bool Path::IsCharacterSafeForShell(char c)
+{
+	if (c >= '0' && c <= '9')
+		return true;
+	if (c >= 'a' && c <= 'z')
+		return true;
+	if (c >= 'A' && c <= 'Z')
+		return true;
+	if (c == '-' || c == '/')
+		return true;
+	return false;
+}
+
+bool Path::EscapeCmdLine(std::string* arg)
+{
+#ifdef _WIN32
+	if (arg->empty() || arg->find_first_of(" \t\"") == std::string::npos)
+		return true;
+
+	std::string temp;
+	temp.reserve(arg->length() + 10);
+	temp += '"';
+
+	for (auto it = arg->begin();; ++it)
+	{
+		int backslash_count = 0;
+		while (it != arg->end() && *it == '\\')
+		{
+			++it;
+			++backslash_count;
+		}
+
+		if (it == arg->end())
+		{
+			temp.append(backslash_count * 2, '\\');
+			break;
+		}
+
+		if (*it == '"')
+		{
+			temp.append(backslash_count * 2 + 1, '\\');
+			temp += '"';
+		}
+		else
+		{
+			temp.append(backslash_count, '\\');
+			temp += *it;
+		}
+	}
+
+	temp += '"';
+	*arg = std::move(temp);
+	return true;
+#else
```

Comment:
This logic doesn't escape backticks, which bash uses to execute commands and collect the result. For example if you put this in one of the fields it isn't escaped correctly:

\`kcalc\`

I think you should consider using single quotes here instead of double quotes, since [for single quotes you only need to escape single quotes](https://www.gnu.org/software/bash/manual/html_node/Single-Quotes.html) (you can use replace with '"'"'). 

Also if you want to disable quoting for simple arguments, I would whitelist characters instead of blacklisting them.